devito/ir/clusters/graph.py

```diff
@@ -4,40 +4,29 @@
 from cached_property import cached_property
 
 from devito.dimension import Dimension
-from devito.symbolics import (Eq, as_symbol, retrieve_indexed, retrieve_terminals,
-                              convert_to_SSA, q_inc, q_indirect, q_timedimension)
+from devito.ir.equations import ClusterizedEq
+from devito.symbolics import (as_symbol, retrieve_indexed, retrieve_terminals,
+                              convert_to_SSA, q_indirect, q_timedimension)
 from devito.tools import DefaultOrderedDict, flatten, filter_ordered
 
 __all__ = ['FlowGraph']
 
 
-class Node(Eq):
+class Node(ClusterizedEq):
```

Comment:
note for reviewers: So much nicer now, as it leverages the existing IR Equation hierarchy